File: SMO/DatabaseSpace/CS/DatabaseSpace/DatabaseSpace.cs
/*============================================================================
  File:    DatabaseSpace.cs 

  Summary: Implements a sample SMO SQL Server database space utility in C#.

  Date:    August 29, 2005
------------------------------------------------------------------------------
  This file is part of the Microsoft SQL Server Code Samples.

  Copyright (C) Microsoft Corporation.  All rights reserved.

  This source code is intended only as a supplement to Microsoft
  Development Tools and/or on-line documentation.  See these other
  materials for detailed information regarding Microsoft code samples.

  THIS CODE AND INFORMATION ARE PROVIDED "AS IS" WITHOUT WARRANTY OF ANY
  KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
  IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
  PARTICULAR PURPOSE.
============================================================================*/
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using Microsoft.SqlServer.Management.Common;
using Microsoft.SqlServer.Management.Smo;
using Microsoft.SqlServer.MessageBox;

namespace Microsoft.Samples.SqlServer
{
    public partial class DatabaseSpace : Form
    {
        // Use the Server object to connect to a specific server
        private Server SqlServerSelection;

        public DatabaseSpace()
        {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e)
        {
            ServerConnection ServerConn;
            ServerConnect scForm;
            DialogResult dr;

            // Display the main window first
            this.Show();
            Application.DoEvents();

            ServerConn = new ServerConnection();
            scForm = new ServerConnect(ServerConn);
            dr = scForm.ShowDialog(this);
            if ((dr == DialogResult.OK) &&
                (ServerConn.SqlConnectionObject.State == ConnectionState.Open))
            {
                SqlServerSelection = new Server(ServerConn);
                if (SqlServerSelection != null)
                {
                    this.Text = Properties.Resources.AppTitle
                        + SqlServerSelection.Name;

                    // Refresh database space list
                    ListDatabaseSpace();
                }
            }
            else
            {
                this.Close();
            }
        }

        private void ListDatabaseSpace()
        {
            try
            {
                this.dataGridView1.Rows.Clear();

                foreach (Database db in SqlServerSelection.Databases)
                {
                    if (db.IsSystemObject == false)
                    {
                        if (db.LogFiles.Count > 0)
                        {
                            this.dataGridView1.Rows.Add(new object[] { db.Name, 
                                db.Size, db.SpaceAvailable / 1024.0, 
                                db.LogFiles[0].Size / 1024.0, 
                                db.LogFiles[0].UsedSpace / 1024.0 });
                        }
                        else
                        {
                            this.dataGridView1.Rows.Add(new object[] { db.Name, 
                                db.Size, db.SpaceAvailable / 1024.0, 
                                0.0, 0.0 });
                        }
                    }
                }
            }
            catch (SmoException ex)
            {
                ExceptionMessageBox emb = new ExceptionMessageBox(ex);
                emb.Show(this);
            }
        }
    }
}

File: SMO/DatabaseSpace/VB/DatabaseSpace/DatabaseSpace.vb
'============================================================================
'  File:    DatabaseSpace.vb 
'
'  Summary: Implements a sample SMO SQL Server database space utility in VB.NET.
'
'  Date:    August 29, 2005
'------------------------------------------------------------------------------
'  This file is part of the Microsoft SQL Server Code Samples.
'
'  Copyright (C) Microsoft Corporation.  All rights reserved.
'
'  This source code is intended only as a supplement to Microsoft
'  Development Tools and/or on-line documentation.  See these other
'  materials for detailed information regarding Microsoft code samples.
'
'  THIS CODE AND INFORMATION ARE PROVIDED "AS IS" WITHOUT WARRANTY OF ANY
'  KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
'  IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
'  PARTICULAR PURPOSE.
'============================================================================
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports Microsoft.SqlServer.Management.Common
Imports Microsoft.SqlServer.Management.Smo
Imports Microsoft.SqlServer.MessageBox

Partial Public Class DatabaseSpace 'The Partial modifier is only required on one class definition per project.
    Inherits Form
    ' Use the Server object to connect to a specific server
    Private SqlServerSelection As Server

    Public Sub New()
        InitializeComponent()
    End Sub

    Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles Me.Load
        Dim ServerConn As ServerConnection
        Dim scForm As ServerConnect
        Dim dr As DialogResult

        ' Display the main window first
        Me.Show()
        Application.DoEvents()

        ServerConn = New ServerConnection()
        scForm = New ServerConnect(ServerConn)
        dr = scForm.ShowDialog(Me)
        If dr = Windows.Forms.DialogResult.OK _
            AndAlso ServerConn.SqlConnectionObject.State = ConnectionState.Open Then
            SqlServerSelection = New Server(ServerConn)
            If Not (SqlServerSelection Is Nothing) Then
                Me.Text = My.Resources.AppTitle + SqlServerSelection.Name

                ' Refresh database space list
                ListDatabaseSpace()
            End If
        Else
            Me.Close()
        End If
    End Sub

    Private Sub ListDatabaseSpace()
        Try
            Me.dataGridView1.Rows.Clear()

            Dim db As Database
            For Each db In SqlServerSelection.Databases
                If db.IsSystemObject = False Then
                    If db.LogFiles.Count > 0 Then
                        Me.dataGridView1.Rows.Add(New Object() {db.Name, db.Size, db.SpaceAvailable / 1024.0, db.LogFiles(0).Size / 1024.0, db.LogFiles(0).UsedSpace / 1024.0})
                    Else
                        Me.dataGridView1.Rows.Add(New Object() {db.Name, db.Size, db.SpaceAvailable / 1024.0, 0.0, 0.0})
                    End If
                End If
            Next db
        Catch ex As SmoException
            Dim emb As New ExceptionMessageBox(ex)
            emb.Show(Me)
        End Try
    End Sub
End Class